### ISSUE TITLE
Liston wind speed and direction downscaling algorithm does not include a curvature length scale

### ISSUE THREAD

**Issue by EricKeenan** [snowpack-model/snowpack#20]
Quick fix for the described issue
```diff
diff --git a/Source/meteoio/meteoio/dataClasses/DEMObject.cc b/Source/meteoio/meteoio/dataClasses/DEMObject.cc
index 2c3cf2a..f8eb452 100644
--- a/Source/meteoio/meteoio/dataClasses/DEMObject.cc
+++ b/Source/meteoio/meteoio/dataClasses/DEMObject.cc
@@ -431,6 +431,8 @@ void DEMObject::CalculateAziSlopeCurve(slope_type algorithm) {
 //This computes the slope and the aspect at a given cell as well as the x and y components of the normal vector
 	double A[4][4]; //table to store neigbouring heights: 3x3 matrix but we want to start at [1][1]
 	                //we use matrix notation: A[y][x]
+	double A2[4][4]; //table to store neigbouring heights: 3x3 matrix but we want to start at [1][1]
+	                //we use matrix notation: A[y][x]
 	if (algorithm==DFLT) {
 		algorithm = dflt_algorithm;
 	}
@@ -467,10 +469,11 @@ void DEMObject::CalculateAziSlopeCurve(slope_type algorithm) {
 				}
 			} else {
 				getNeighbours(i, j, A);
+				getNeighbours2(i, j, A2);
 				double new_slope, new_Nx, new_Ny, new_Nz;
 				(this->*CalculateSlope)(A, new_slope, new_Nx, new_Ny, new_Nz);
 				const double new_azi = CalculateAzimuth(new_Nx, new_Ny, new_Nz, new_slope);
-				const double new_curvature = getCurvature(A);
+				const double new_curvature = getCurvature(A2);
 				if (update_flag&SLOPE) {
 					slope(i,j) = new_slope;
 					azi(i,j) = new_azi;
@@ -782,6 +785,34 @@ void DEMObject::getNeighbours(const size_t& i, const size_t& j, double A[4][4])
 	}
 }
 
+void DEMObject::getNeighbours2(const size_t& i, const size_t& j, double A[4][4]) const
+{ //this fills a 3x3 table containing the neighboring values
+	const int n_radius = 10; // Number of grid cells which contain the topographic slope curvature length scale.
+
+	if ((i>n_radius-1 && i<(getNx()-n_radius)) && (j>n_radius-1 && j<(getNy()-n_radius))) {
+		//this is the normal case
+		A[1][1] = grid2D(i-n_radius, j+n_radius);
+		A[1][2] = grid2D(i, j+n_radius);
+		A[1][3] = grid2D(i+n_radius, j+n_radius);
+		A[2][1] = grid2D(i-n_radius, j);
+		A[2][2] = grid2D(i, j);
+		A[2][3] = grid2D(i+n_radius, j);
+		A[3][1] = grid2D(i-n_radius, j-n_radius);
+		A[3][2] = grid2D(i, j-n_radius);
+		A[3][3] = grid2D(i+n_radius, j-n_radius);
+	} else { //somewhere on the border
+		A[1][1] = safeGet((signed)i-n_radius, (signed)j+n_radius);
+		A[1][2] = safeGet((signed)i, (signed)j+n_radius);
+		A[1][3] = safeGet((signed)i+n_radius, (signed)j+n_radius);
+		A[2][1] = safeGet((signed)i-n_radius, (signed)j);
+		A[2][2] = safeGet((signed)i, (signed)j);
+		A[2][3] = safeGet((signed)i+n_radius, (signed)j);
+		A[3][1] = safeGet((signed)i-n_radius, (signed)j-n_radius);
+		A[3][2] = safeGet((signed)i, (signed)j-n_radius);
+		A[3][3] = safeGet((signed)i+n_radius, (signed)j-n_radius);
+	}
+}
+
 double DEMObject::safeGet(const int& i, const int& j) const
 {//this function would allow reading the value of *any* point,
 //that is, even for coordinates outside of the grid (where it would return nodata)
diff --git a/Source/meteoio/meteoio/dataClasses/DEMObject.h b/Source/meteoio/meteoio/dataClasses/DEMObject.h
index 3d1fd31..0e1c298 100644
--- a/Source/meteoio/meteoio/dataClasses/DEMObject.h
+++ b/Source/meteoio/meteoio/dataClasses/DEMObject.h
@@ -146,6 +146,7 @@ class DEMObject : public Grid2DObject {
 		static void surfaceGradient(double& dx_sum, double& dy_sum, double A[4][4]);
 		static double avgHeight(const double& z1, const double &z2, const double& z3);
 		void getNeighbours(const size_t& i, const size_t& j, double A[4][4]) const;
+		void getNeighbours2(const size_t& i, const size_t& j, double A[4][4]) const;
 		double safeGet(const int& i, const int& j) const;
 
 		double max_shade_distance; ///< maximum distance to look for when computing a horizon
```

**Comment by EricKeenan** [comment]
@nwever Is it fair to say this issue is closed?

**Comment by nwever** [comment]
Indeed! Fully fixed by commit de40c8325735e97053a4cd092a7824bfd83adb99

ETA can now be specified in the ListonWind algorithm to specify the curvature length scale in m.
